### DISCUSSION TITLE
was cancelled because it was hanging in between 2 steps

### DISCUSSION THREAD

**Discussion by FrancisVarga** [windmill-labs/windmill#4492] [Questions]
Ill get this a lot when ill run parallel tasks. any hints how to solve it?

**Comment by rubenfiszel** [comment]
Hi, 

Could you share a minimal flow that expose this behavior + details about your setup please (number of workers, windmill version, infra)

It usually means flow state  transitions are not happening or not in time at the very least

**Reply by FrancisVarga** [reply]
Looks like this:
![image](https://github.com/user-attachments/assets/3dfb4e3e-ec34-435d-a6b0-462f896277f7)

What it does since its hanging im generating batches which loops over and the batches are looping over againg for the items. the issue is if parallel execution is quite big its hanging.

runs on docker compose with 32gb ram machine the postgres runs on 64gb ram both servers have 16 cores. the it runs with 5 workers servers with 10 workers.

**Reply by rubenfiszel** [reply]
You can export the full flow yaml directly. Please send us a minimal reproduction that we can run fully of a flow that expose this behavior and send us the yaml directly (you can replace steps that take time with a python script):
```
import time

def main():
  time.sleep(x)
  ```
